docs/cmake-project-view.md

```diff
@@ -0,0 +1,116 @@
+ 
+To make options visible in the status bar take up less space, you can configure certain options to be only the respective icon through the `icon` option in `statusBarVisibility` or specify a given character length through the `compact` option in `statusBarVisibility`. If a `statusBarVisibility` option is set to `compact` you can then specify an integer length for how many characters you want an option to take up through the `statusBarLength` option. It will truncate your existing status bar option to that specified character length. The default option is 20 characters
+ 
+The full level of options for the CMake status can be seen below:
+
+### CMake Status Bar Configuration Options in your Settings Json
+```json
+"cmake.status.statusBarVisibility": "visibile", "icon", "compact", "hidden" 
+"cmake.status.advanced": { 
+
+        "folder": { 
+        	"statusBarVisibility": "visible", "icon", "compact", "hidden" 
+            "statusBarLength": 20
+            "projectStatusVisibility": "visible", "hidden" 
+        }, 
+        "configure": { 
+            	"projectStatusVisibility": "visible", "hidden" 
+        }, 
+        "configurePreset": { 
+            	"statusBarVisibility": "visible", "icon", "compact", "hidden" 
+            	"statusBarLength": 20
+        }, 
+        "kit": { 
+            	"statusBarVisibility": "visible", "icon", "compact", "hidden" 
+            	"statusBarLength": 20 
+        }, 
+        "variantStatus": { 
+            	"statusBarVisibility": "visible", "icon", "compact", "hidden" 
+        }, 
+        "build": { 
+            	"statusBarVisibility": "visible", "icon", "hidden" 
+            	"projectStatusVisibility": "visible", "hidden" 
+        }, 
+        "buildPreset": { 
+            	"statusBarVisibility": "visible", "icon", "compact", "hidden" 
+            	"statusBarLength": 20 
+        }, 
+        "buildTarget": { 
+            	"statusBarVisibility": "visible", "compact", "hidden" 
+            	"statusBarLength": 20 
+        }, 
+        "ctest": { 
+            	"statusBarVisibility": "visible", "icon", "compact", "hidden" 
+            	"statusBarLength": 20 
+            	"color": true, false 
+            	"projectStatusVisibility": "visible", "hidden" 
+        }, 
+        "testPreset": { 
+            	"statusBarVisibility": "visible", "icon", "compact", "hidden" 
+            	"statusBarLength": 20
+        },
+        "launchTarget": { 
+            	"statusBarVisibility": "visible", "compact", "hidden" 
+            	"statusBarLength": 20 
+        }, 
+        "debug": { 
+            	"statusBarVisibility": "visible", "hidden" 
+            	"projectStatusVisibility": "visible", "hidden" 
+        },
+        "launch": { 
+            	"statusBarVisibility": "visible", "hidden" 
+            	"projectStatusVisibility": "visible", "hidden" 
+        }
+}
+```
+
```

Comment:
I think a note saying which settings apply for the Project Status View would be helpful, and a clarification on the children of the parent node being affected as well by the visibility setting

Reply:
This would be folder, configure, build, launch, debug, and ctest